src/components/job/JobList.jsx

```diff
@@ -0,0 +1,48 @@
+import { useRequest } from "ahooks";
+import { useEffect, useMemo } from "react";
+import { Link, useParams } from "react-router-dom";
+
+import { useAuth, useJob } from "@/models";
+
+export default function JobList() {
+  const { listJobs } = useJob();
+  const { jobId } = useParams();
+  const { run, data } = useRequest(async () => listJobs(), {
```

Comment:
Eventually the ability to fetch QueryConstraint be added to useAuth to control what output list each other gets?

Reply:
the user account will limit what he can see. like a client cannot see pending jobs

Of what he can see, we may still still need a filter. like he want to see jobs with keyword `counter`, that is what `queryConstraints` does.

Reply:
So the user account will contribute parameters to queryConstraints? Or will there be separate jobList methods for each type of user?

Reply:
no.
the account decides the whole collection of jobs he can see.
then use query to filter on the limited collection

Reply:
Oh I see, thanks!